src/backend/distributed/executor/multi_utility.c

```diff
@@ -1045,6 +1051,33 @@ ErrorIfDistributedAlterSeqOwnedBy(AlterSeqStmt *alterSeqStmt)
 
 
 /*
+ * ErrorIfUnsupportedTruncateStmt errors out if the command attempts to
+ * truncate a distributed foreign table.
+ */
+static void
+ErrorIfUnsupportedTruncateStmt(TruncateStmt *truncateStatement)
```

Comment:
If foreign table belongs to an extension it may choose to implement truncate command. 

cstore_fdw executes truncate command internally, and does not let PostgreqSql to  error out here.